app/src/main/java/br/com/lucas/pomodoroapp/helpers/AlarmManagerHelper.kt

```diff
@@ -0,0 +1,50 @@
+package br.com.lucas.pomodoroapp.helpers
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.SystemClock
+import androidx.core.app.AlarmManagerCompat
+import br.com.lucas.pomodoroapp.core.receiver.AlarmReceiver
+import br.com.lucas.pomodoroapp.database.Task
+import dagger.hilt.android.internal.Contexts.getApplication
+import java.util.concurrent.TimeUnit
+
+class AlarmManagerHelper(private val context: Context) {
+    fun setExactAlarm(
+        task: Task
+    ) {
+        buildAlarmManager()?.let { manager ->
+            AlarmManagerCompat.setExactAndAllowWhileIdle(
+                manager,
+                AlarmManager.ELAPSED_REALTIME_WAKEUP,
+                SystemClock.elapsedRealtime() +
+                        TimeUnit.MINUTES.toMillis(task.taskMinutes * 1L),
+                getPendingIntent(task)
+            )
+        }
+    }
+
+    private fun buildAlarmManager(): AlarmManager? =
```

Comment:
Instead of a method you could have `AlarmManager` as a global property for this class, something like:

```kotlin
class AlarmManagerHelper(context: Context) {
          private val alarmManager: AlarmManager = context.getSystemService(
                     Context.ALARM_SERVICE
           )
           ...
}
```